chore(walletconnect)_: various improvements applied

saledjenic opened the pull request:
These changes are the second step in cleaning the wallet API and moving the things to the place where they logically belong.

- `WalletConnectTransfer` identified as a new transfer type
- Wallet-related endpoints that logically belong to the wallet moved from the wallet connect service
- Wallet connect service now receives `transfer.TransactionManager` instead of `transactions.Transactor`
- Deadlock issue when trying to send the tx with the wrong nonce fixed

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (4)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_multiplication_x: | 81e62983 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4414/1/) | 2023-12-01 15:59:26 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4414/1//consoleText) |
| :heavy_check_mark: | 81e62983 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4414/1/) | 2023-12-01 16:00:21 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231201-155646-81e629-pr4414.zip) |
| :heavy_check_mark: | 81e62983 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4414/1/) | 2023-12-01 16:01:28 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231201-155645-81e629-pr4414.zip) |
| :heavy_check_mark: | 81e62983 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4414/1/) | 2023-12-01 16:02:04 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231201-155645-81e629-pr4414.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | fc59cb58 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4414/2/) | 2023-12-01 16:44:45 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231201-164258-fc59cb-pr4414.zip) |
| :heavy_check_mark: | fc59cb58 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4414/2/) | 2023-12-01 16:48:15 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231201-164255-fc59cb-pr4414.zip) |
| :heavy_check_mark: | fc59cb58 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4414/2/) | 2023-12-01 16:48:56 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231201-164255-fc59cb-pr4414.aar) |
| :heavy_check_mark: | fc59cb58 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4414/2/) | 2023-12-01 17:17:28 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4414/2//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 0ed1ee9e | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-4414/3/) | 2023-12-01 16:46:59 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-231201-164535-0ed1ee-pr4414.zip) |
| :heavy_check_mark: | 0ed1ee9e | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-4414/3/) | 2023-12-01 16:50:41 | ~1 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-231201-164859-0ed1ee-pr4414.aar) |
| :heavy_check_mark: | 0ed1ee9e | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-4414/3/) | 2023-12-01 16:51:27 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-231201-164818-0ed1ee-pr4414.zip) |
| :heavy_check_mark: | 0ed1ee9e | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4414/3/) | 2023-12-01 17:51:35 | ~34 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-4414/3//consoleText) |

saledjenic committed:
chore(walletconnect)_: various improvements applied

- `WalletConnectTransfer` identified as a new transfer type
- Wallet-related endpoints that logically belong to the wallet moved from the wallet connect service
- Wallet connect service now receives `transfer.TransactionManager` instead of `transactions.Transactor`
- Deadlock issue when trying to send the tx with the wrong nonce fixed

stefandunca reviewed (approved):
Nice simplification
